src/SystemCommands-MessageCommands/SycHierarchicalSendersCommand.class.st

```diff
@@ -0,0 +1,45 @@
+"
+This command find the senders of a given method in the classes upper in the inheritance chain above a given method.
+"
+Class {
+	#name : 'SycHierarchicalSendersCommand',
+	#superclass : 'SycChangeMessageSignatureCommand',
+	#category : 'SystemCommands-MessageCommands',
+	#package : 'SystemCommands-MessageCommands'
+}
+
+{ #category : 'execution' }
+SycHierarchicalSendersCommand >> applyResultInContext: a [
+]
+
+{ #category : 'accessing' }
+SycHierarchicalSendersCommand >> defaultMenuIconName [
+
+	^ #edit
+]
+
+{ #category : 'accessing' }
+SycHierarchicalSendersCommand >> defaultMenuItemName [
+
+	^ 'Hierarchical senders'
+]
+
+{ #category : 'execution' }
+SycHierarchicalSendersCommand >> execute [
+
+	Smalltalk tools messageList browse: self searchInTheWholeHierarchy.
```

Comment:
You should not implement it from scratch really. 
Calypso already provides a model for such queries. 
Just see how ClyShowLocalImplementorsCommand is implemented. It's really what you want here but for implementors.

Reply:
I implemented it this way:

[ClyShowMessageLocalSenderCommand.txt](https://github.com/pharo-project/pharo/files/13733054/ClyShowMessageLocalSenderCommand.txt)

Reply:
I want to use the command without depending on Calypso, is that possible with your implementation?

Reply:
Well if you don't use query model of Calypso you loose a lot of functionality and you duplicate the query logic which is already implemented.
For example:
- there is no indication that the message browser is in a scoped mode (scope of a hierarchy)
- there is no auto update when new senders appears in the system

Reply:
And more interesting is scopes in spawned message browser and tab based navigation when you request local sender from another message browser.
For example try open Inheritance for any method. This is analog of local senders but for implementors. You can see that you can see that the current scope is "Hierarchy" and you can switch it back to full image or other options.
With your approach the message browser just shows given list of methods. No scopes are available.

Reply:
Thanks for pointing out, I will open another issue to check those issues you mention.